include/net/vlan.h

```diff
@@ -0,0 +1,76 @@
+/** @file
+ * @brief VLAN specific definitions.
```

Comment:
VLAN is an ethernet extension. Just move this in ethernet.h (and all the functions below are using net_eth_ so it's really better to have these in ethernet.h)

Reply:
That would cause circular dependency with net_pkt.h, that is why I had to create separate header file for small subset of things that manipulate vlan header.

Reply:
Well, don't inline the vlan net_pkt function then. That will fix the issue.

Reply:
Hmm, I am not really following you here. Currently ethernet.h needs to include net_pkt.h so we cannot include ethernet.h in net_pkt.h. So by putting couple of vlan header manipulator functions to vlan.h, this issue can be mitigated.

Reply:
These functions your are introducing in net_pkt.h: make them not inline, in net_pkt.c it should be ok to include ethernet.h

Reply:
Sure, that is one option.  I would prefer that we keep the inline versions as these are called for almost every incoming packet.

Reply:
I don't like mixing net_eth functiont in net_pkt.  But ok, net_pkt is going to be un-exposed from user someday hopefully.

Just rename vlan.h to ethernet_vlan.h, and the function inside prefixed net_eth_vlan_ it's clearer then.